bondcpp/CMakeLists.txt

```diff
@@ -1,41 +1,77 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(bondcpp)
 
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_ros REQUIRED)
 find_package(Boost REQUIRED)
-find_package(catkin REQUIRED bond cmake_modules roscpp smclib)
-find_package(UUID REQUIRED)
+find_package(ament_cmake_auto REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(rcl REQUIRED)
+find_package(rcl_interfaces REQUIRED)
+find_package(rmw REQUIRED)
+find_package(rmw_implementation REQUIRED)
+find_package(PkgConfig REQUIRED)
+find_package(Threads REQUIRED)
+
+pkg_search_module(UUID REQUIRED uuid)
+
+find_package(rcutils REQUIRED)
+find_package(smclibcpp REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(bond REQUIRED)
 
-include_directories(
+set(INCLUDE_DIRS
   include
-  ${catkin_INCLUDE_DIRS}
-  ${Boost_INCLUDE_DIRS}
+  ${smclibcpp_INCLUDE_DIRS}
+  ${rclcpp_INCLUDE_DIRS}
+  ${bond_INCLUDE_DIRS}
   ${UUID_INCLUDE_DIRS}
-)
+  ${rcl_INCLUDE_DIRS})
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS bond roscpp smclib
-  DEPENDS Boost
-)
+set(LIBS
+  ${rclcpp_LIBRARIES}
+  ${smclibcpp_LIBRARIES}
+  ${bond_LIBRARIES}
+  ${rcl_LIBRARIES}
+  ${rcutils_LIBRARIES})
+
+include_directories(${INCLUDE_DIRS})
```

Comment:
A bunch of these ${INCLUDE_DIRS} and ${LIBS} can be eliminated by using `ament_target_dependencies`

Reply:
@mjcarroll 
Not sure how to use,
can you give example?